CMakeLists.txt

```diff
@@ -325,6 +325,16 @@ if(CUDA_FOUND)
     option(COMPILE_AMPERE  "Compile GPU version with SM80 support" ON)
     LIST(APPEND COMPUTE -Wno-deprecated-gpu-targets)
   endif()
+  if(CUDA_VERSION VERSION_EQUAL "11.1" OR CUDA_VERSION VERSION_GREATER "11.1")
+    option(COMPILE_KEPLER      "Compile GPU version with SM35 support" OFF) # deprecated for CUDA 11
+    option(COMPILE_MAXWELL     "Compile GPU version with SM50 support" OFF) # deprecated for CUDA 11
+    option(COMPILE_PASCAL      "Compile GPU version with SM60 support" ON)
+    option(COMPILE_VOLTA       "Compile GPU version with SM70 support" ON)
+    option(COMPILE_TURING      "Compile GPU version with SM75 support" ON)
+    option(COMPILE_AMPERE      "Compile GPU version with SM80 support" ON)
+    option(COMPILE_AMPERE_RTX  "Compile GPU version with SM86 support" ON)
```

Comment:
It seems that all options are defined twice for CUDA 11.1? To avoid code duplication, can you just add an if statement defining COMPILE_AMPERE_RTX only for CUDA 11.1+ into the if-block for CUDA 11+ above? Alternatively, a condition checking that CUDA is lower than 11.1 could be added into the first if statement.

Reply:
So you want something like
```
if(CUDA_VERSION VERSION_EQUAL "11.0" OR CUDA_VERSION VERSION_GREATER "11.0")
    option(COMPILE_KEPLER  "Compile GPU version with SM35 support" OFF) # deprecated for CUDA 11
    option(COMPILE_MAXWELL "Compile GPU version with SM50 support" OFF) # deprecated for CUDA 11
    option(COMPILE_PASCAL  "Compile GPU version with SM60 support" ON)
    option(COMPILE_VOLTA   "Compile GPU version with SM70 support" ON)
    option(COMPILE_TURING  "Compile GPU version with SM75 support" ON)
    option(COMPILE_AMPERE  "Compile GPU version with SM80 support" ON)
    if (CUDA_VERSION VERSION_EQUAL "11.1" OR CUDA_VERSION VERSION_GREATER "11.1")
      option(COMPILE_AMPERE_RTX  "Compile GPU version with SM86 support" ON)
    endif()
    LIST(APPEND COMPUTE -Wno-deprecated-gpu-targets)
  endif()
  ```
  ?

Reply:
tbh, we can do nested ifs for older versions as well

Reply:
Or adding `AND CUDA_VERSION VERSION_LESS "11.1"` to the first if-block for CUDA 11, or removing all except COMPILE_AMPERE_RTX from the last if-block?

Reply:
Oh, I noticed that previous blocks repeat option definitions anyway, so feel free to ignore, we may clean this up later.

Reply:
It's going to be a lot of nested ifs... Shall we merge this for now "as is" and in the future, if/when we do refactoring, we can fix it?